src/traccuracy/matchers/_iou.py

```diff
@@ -98,29 +98,36 @@ def match_iou(gt, pred, threshold=0.6):
     return mapper
 
 
-class IOUMatched(Matched):
-    def __init__(self, gt_graph, pred_graph, iou_threshold=0.6):
+class IOUMatcher(Matcher):
+    def __init__(self, iou_threshold=0.6):
         """Constructs a mapping between gt and pred nodes using the IoU of the segmentations
 
         Lower values for iou_threshold will be more permissive of imperfect matches
 
         Args:
-            gt_graph (TrackingGraph): TrackingGraph for the ground truth with segmentations
-            pred_graph (TrackingGraph): TrackingGraph for the prediction with segmentations
             iou_threshold (float, optional): Minimum IoU value to assign a match. Defaults to 0.6.
+        """
+        self.iou_threshold = iou_threshold
+
+    def _compute_mapping(self, gt_graph: "TrackingGraph", pred_graph: "TrackingGraph"):
+        """Computes IOU mapping for a set of grpahs
+
+        Args:
+            gt_graph (TrackingGraph): Tracking graph object for the gt with segmentation data
+            pred_graph (TrackingGraph): Tracking graph object for the pred with segmentation data
 
         Raises:
             ValueError: Segmentation data must be provided for both gt and pred data
-        """
-        self.iou_threshold = iou_threshold
 
+        Returns:
+            Matched: Matched data object containing IOU mapping
+        """
         # Check that segmentations exist in the data
         if gt_graph.segmentation is None or pred_graph.segmentation is None:
             raise ValueError(
                 "Segmentation data must be provided for both gt and pred data"
             )
 
-        super().__init__(gt_graph, pred_graph)
+        mapping = match_iou(gt_graph, pred_graph, threshold=self.iou_threshold)
```

Comment:
Would you get rid of the copy in the `Matched` class? My only hesitation is keeping it in the Matched class ensures that it will always happen even if we have someone implementing a new `Matcher`. I'm worried about making two sets of copies as we get into larger graphs.

Reply:
I found a place to move it into the `Matcher` class which I think makes the most sense.

Reply:
Thanks, lgtm.